[Cherry pick] Change worker-setup-hook -> worker process setup hook (#37463)
ray-project/ray: doc/source/ray-observability/user-guides/configure-logging.md

```{admonition} Caution
:class: caution
This is an experimental feature. The semantic of the API is subject to change.
It doesn't support [Ray Client](ray-client-ref) yet.

> Currently, all the runtime environment passed to a driver (`ray.init(runtime_env={...})`) will be ignored if you specify any runtime environment via [Ray Job Submission](jobs-quickstart) API (`ray job submit --working-dir` or `ray job submit --runtime-env`).

Reviewer: ```suggestion
All runtime environment settings passed to a driver (`ray.init(runtime_env={...})`) are ignored when you specify any runtime environment using [Ray Job Submission](jobs-quickstart) API (`ray job submit --working-dir` or `ray job submit --runtime-env`).
```
